src/filesystem/impls/filer/FilerFileSystem.js

```diff
@@ -304,14 +305,20 @@ define(function (require, exports, module) {
 
                     // If this is a big image (>250K), resize it first
                     if(Content.isResizableImage(ext) && Content.isImageTooLarge(data.length)) {
-                        ImageResizer.resize(path, data, function(err, resized) {
-                            if(err) {
-                                return callback(err);
-                            }
-
-                            data = resized;
+                        if(path.match(".*\.gif")&& isAnimated(data)) {
```

Comment:
Let's move this logic (i.e., checking if this is both a `.gif` and whether it's animated or not) into `Content.isResizableImage()`, so that you don't need to do any extra `if` checks here.

Reply:
@humphd Ohhhh. It's nice. Thank you!

Reply:
One further thought, you'll either need to modify what `isResizableImage()` takes as an argument--it uses an extension now, but you'd need the entire path.  Or you could split this into two functions, and have the `isAnimated()` be a separate function that lives on `Content`.  Does that make sense?

Reply:
@humphd I need to bring data into Content.isResizableImage() for animation check, but if I just I add "data" argument, I got a data type error. To avoid this, do I need the entire path? Is that what you mean?

It seems this one works to filter animated gif and png, but is that not nice for jpg and jpeg?
// FillerFileSystem.js line307 (added isAnimated check at the top of the original)
<img width="586" alt="snapcrab_noname_2018-1-10_19-22-22_no-00" src="https://user-images.githubusercontent.com/22462471/34802444-aac3c646-f63b-11e7-9c2a-afbd0b4c07c6.png">

Reply:
Yes, I'd pass in the whole path to a new `isAnimated` method on `Content`:

```js
if((Content.isResizableImage(ext) && !Content.isAnimated(path)) && Content.isImageTooLarge(data.length)) {
...
```

Reply:
Oh, that's why I can have two functions...! 
I see. Thank you!

Reply:
@humphd 
I'd like to push the latest one,
but could you let me ask about the rebase please?

++now in #2307branch
   git pull --rebase origin master
++fix conflicts
   git add *
   git rebase --continue
   git push origin #2307branch -f
   
Is that correct way?
Should I do "rebase" now, or after the whole fixing was approved?

Reply:
I would do this:

```
git checkout master
git pull origin master
git checkout #2307branch
git rebase master
++fix conflicts
git add <files-you-changed-one-by-one>
git rebase --continue
git push origin #2307branch -f
```

Also, I'd recommend you rebase when you've finished everything and gotten a positive review.  Though you can do it either way.

Reply:
@humphd Thank you so much! Then, I will do rebase later!